Source/GSXibKeyedUnarchiver.m

```diff
@@ -143,8 +165,46 @@ - (NSData *) _preProcessXib: (NSData *)data
-                          [customClassDict setObject:[[xmlObjs objectAtIndex:index] stringValue] forKey:key];
+                          NSString *cn = [[xmlObjs objectAtIndex: index] stringValue]; // className
+
+                          [customClassDict setObject:cn forKey:key];
+                                                    
+                          //
+                          // If we are in IB/Gorm build the custom classes map so that we don't instantiate
+                          // classes which don't exist (yet) in IB/Gorm.  This allows editing of the model
+                          // in IB/Gorm.  If we are in the live app, don't bother as it's a waste of memory.
+                          //
+                          if ([NSClassSwapper isInInterfaceBuilder] == YES)
+                            {
+                              NSString *xpath = [NSString stringWithFormat: @"//object[@class=\"IBClassDescriber\"]"
+                                                          @"//string[@key=\"className\"][text()=\"%@\"]"
+                                                          @"/../string[@key=\"superclassName\"]", cn];
+                              NSArray *descriptionObjs = [document nodesForXPath: xpath error: NULL];
+                              if ([descriptionObjs count] > 0)
+                                {
+                                  NSUInteger idx = [key rangeOfString: @"."].location;
+                                  if (idx != NSNotFound) // unlikely to be NSNotFound...
+                                    {
+                                      NSString *num = [key substringToIndex: idx];
+                                      NSXMLNode *descriptionNode = [descriptionObjs objectAtIndex: 0];
+                                      NSString *sc = [descriptionNode stringValue]; // superclassName
+                                      NSString *refXPath = [NSString stringWithFormat:
+                                                                       @"//object[@class=\"IBMutableOrderedSet\"][@key=\"objectRecords\"]"
+                                                                     @"/object/object[@class=\"IBObjectRecord\"]/int[@key=\"objectID\"]"
+                                                                     @"[text()=\"%@\"]/../reference[@key=\"object\"]/@ref", num];
+                                      
+                                      NSArray *refNodes = [document nodesForXPath: refXPath error: NULL];
+                                      if ([refNodes count] > 0)
+                                        {
+                                          NSXMLElement *refNode = [refNodes objectAtIndex: 0];
+                                          NSString *refId = [refNode stringValue];
+                                          NSDictionary *dict = [NSDictionary dictionaryWithObjectsAndKeys:
```

Comment:
You could use the method -createCustomClassRecordForId: withParentClass: forCustomClass: (By moving that method up one class). That way it is easier to change the storage structure we use.

Reply:
Done.